doc/specs/#1595 - Suggestions UI/Snippets.md

```diff
@@ -0,0 +1,526 @@
+
+```pwsh
+$s=Invoke-GitHubGraphQlApi "query{organization(login:`"Microsoft`"){projectV2(number: 159) { id } } }"
+
+$tasks = get-GitHubIssue  -Labels "Issue-Task" -state open
+$bugs = get-GitHubIssue  -Labels "Issue-Bug" -state open
+$issues = $tasks + $bugs
+
+$issues | ? {$_.labels.Name -NotContains "Needs-Triage" } | ? { $_.milestone.title -Ne "Icebox ❄" } | ? type -Ne "PullRequest" | select -expand node_id | % {
+  $resp = Add-GitHubBetaProjectItem -ProjectNodeId $s.organization.projectV2.id -ContentNodeId $_ ;
+}
+```
+
+As just a raw sendInput action with a single `input`, this would look like the following:
+
+```jsonc
+{
+    "command":
+    {
+        "action": "sendInput",
+        "input": "$s=Invoke-GitHubGraphQlApi \"query{organization(login:`\"Microsoft`\"){projectV2(number: 159) { id } } }\"\r\n$tasks = get-GitHubIssue  -Labels \"Issue-Task\" -state open\r\n$bugs = get-GitHubIssue  -Labels \"Issue-Bug\" -state open\r\n$issues = $tasks + $bugs\r\n$issues | ? {$_.labels.Name -NotContains \"Needs-Triage\" } | ? { $_.milestone.title -Ne \"Icebox ❄\" } | ? type -Ne \"PullRequest\" | select -expand node_id | % {\r\n  $resp = Add-GitHubBetaProjectItem -ProjectNodeId $s.organization.projectV2.id -ContentNodeId $_ ;\r\n}"
+    },
+    "name": "Upload to project board",
+    "description": "Sync all our issues and bugs that have been triaged and are actually on the backlog to the big-ol project",
+},
+```
+
+This JSON is basically entirely unusable. Since JSON doesn't support multiline
+strings, then every line has to be joined to a single line, separated by `\r\n`.
+
+Instead, the following version of this command uses an array for the `input`
+parameter. This then implies that each string should be sent in sequence, with
+<kbd>enter</kbd> between them.
+
+```jsonc
+{
+    "command":
+    {
+        "action": "sendInput",
+        "input":
+        [
+            "$s=Invoke-GitHubGraphQlApi \"query{organization(login:`\"Microsoft`\"){projectV2(number: 159) { id } } }\"",
+            "$tasks = get-GitHubIssue  -Labels \"Issue-Task\" -state open",
+            "$bugs = get-GitHubIssue  -Labels \"Issue-Bug\" -state open",
+            "$issues = $tasks + $bugs",
+            "$issues | ? {$_.labels.Name -NotContains \"Needs-Triage\" } | ? { $_.milestone.title -Ne \"Icebox ❄\" } | ? type -Ne \"PullRequest\" | select -expand node_id | % {",
+            "  $resp = Add-GitHubBetaProjectItem -ProjectNodeId $s.organization.projectV2.id -ContentNodeId $_ ;",
+            "}",
+            ""
+        ]
+    },
+    "name": "Upload to project board",
+    "description": "Sync all our issues and bugs that have been triaged and are actually on the backlog to the big-ol project",
+},
+```
+
+This is slightly more maintainable. But now, a user could set
+`"waitForSuccess": true`, and if any part of the script fails, then the rest of
+it won't be sent to the shell.

```

Comment:
I'm considering punting this whole parameter to Future Considerations. It's already dependent on Shell Integration being enabled, and that makes the whole section hard to parse. You've got to wrap basically everything that mentions it with "Assuming the user also has shell integration enabled, then {{this parameter does something}}".

I called it `waitForSuccess`, because it would literally wait for the FTCS_D to indicate the command ended (and the exit code), before sending more text. `stopIfFailed` reads a little different. That wording seems to imply that it would only stop sending commands if it saw an error. (but in the default case, it's not going to be seeing anything, because the average user doesn't have SI enabled).